example/backend/state-transition/src/state-machine.ts

```diff
@@ -0,0 +1,83 @@
+import { PaimaSTM } from "@paima/sm";
+import { grammar } from "@example/data-types";
+import type { BaseStfInput, BaseStfOutput } from "@paima/sm";
+import { insertStateMachineInput } from "@example/database";
+import type { StartConfigGameStateTransitions } from "@paima/runtime";
+// import { createScheduledData } from "@paima/db";
+
+type MyEvents = {}; // TODO: replace
+const stm = new PaimaSTM<typeof grammar, MyEvents>(grammar);
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+stm.addStateTransition(
+  "attack",
+  async (data) => {
+    await sleep(0);
+    return {
```

Comment:
no, we should instead do everything through yields

ex:
- `yield <some db>`
- `yield <some event>`

If users don't want to do yield (although they're getting popular with libs like Effect.ts gaining traction), you can write a promise-api that just forwards to yields as a wrapper (I'm okay with either one)

Reply:
Follow Up in new task PAI-48